src/common/getType.ts

```diff
@@ -485,42 +514,58 @@ function getOpCodeType(
       const t = types[0] as IntegerType;
       if (isFiniteType(t))
         return textType(
-          Math.max(
-            ...[t.low, t.high].map(
-              (x) =>
+          integerTypeIncludingAll(
+            ...[t.low, t.high, ...(typeContains(t, 0n) ? [0n] : [])].map((x) =>
+              BigInt(
                 x.toString(
                   expr.op === "int_to_bin"
                     ? 2
                     : expr.op === "int_to_hex"
                     ? 16
                     : 10
                 ).length
+              )
             )
-          )
+          ),
+          true
         );
-      return textType();
+      return textType(integerType(1), true);
     }
     case "text_to_int": {
-      expectType(textType());
+      expectType(textType(integerType(), true));
       const t = types[0] as TextType;
-      if (t.capacity === Infinity) return integerType();
+      if (!isFiniteType(t.codepointLength)) return integerType();
       return integerType(
-        1n - 10n ** BigInt(t.capacity - 1),
-        10n ** BigInt(t.capacity) - 1n
+        1n - 10n ** (t.codepointLength.high - 1n),
+        10n ** t.codepointLength.high - 1n
       );
     }
     case "bool_to_int":
       expectType(booleanType);
       return integerType(0, 1);
-    case "byte_to_char":
+    case "byte_to_text":
```

Comment:
It's not clear what `byte_to_text` does. `byte_to_char` is a more descriptive name here — maybe `byte_to_char_ascii` to clarify encoding. Similarly, the next builtin could be `int_to_char_codepoint`.

Reply:
A wanted to avoid the slighly ambiguous word character. The output might not be ascii since the input can be > 127. Maybe `byte_to_text_byte`? Not sure what's best here, however, for the second one, `int_to_codepoint` seems very clear to me.

Reply:
I was bikeshedding. `byte_to_text` is good enough. And `int_to_codepoint` is plenty descriptive.